test/unittests/test_dd_synthesis_dc.cpp

```diff
@@ -0,0 +1,51 @@
+#include "algorithms/synthesis/dd_synthesis.hpp"
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestDDSynthDc: public testing::TestWithParam<std::string> {
+protected:
+    TruthTable                     tt{};
+    std::string                    testCircuitsDir = "./circuits/";
+    std::unique_ptr<dd::Package<>> dd              = std::make_unique<dd::Package<>>(15U);
+    std::string                    fileName;
+
+    void SetUp() override {
+        fileName = testCircuitsDir + GetParam() + ".pla";
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(TestDDSynth, TestDDSynthDc,
+                         testing::Values(
+                                 "huff_1",
+                                 "dcX2bit",
+                                 "dc3bit",
+                                 "dc3bitNew",
+                                 "dcX4bit",
+                                 "sym6_32"),
+                         [](const testing::TestParamInfo<TestDDSynthDc::ParamType>& info) {
+                             auto s = info.param;
+                             std::replace( s.begin(), s.end(), '-', '_');
+                             return s; });
+
+TEST_P(TestDDSynthDc, GenericDDSynthesisDcTest) {
+    EXPECT_TRUE(readPla(tt, fileName));
+    extend(tt);
+
+    encodeHuffman(tt);
+
+    const auto ttDD = buildDD(tt, dd);
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    DDSynthesizer synthesizer(tt.nInputs());
+    const auto&   qc = synthesizer.synthesize(ttDD, dd);
+
+    EXPECT_TRUE(!qc.empty());
+
+    std::cout << synthesizer.numGate() << "\n";
+    std::cout << synthesizer.getExecutionTime() << "\n";
```

Comment:
It would be great if these tests would actually test something more than just that the synthesis per se works.

Reply:
The encoded `ttDD` is know only after undergoing the synthesis. How do you suggest we test in this type of situation?

Reply:
Yeah. And right now, without proper tests, the circuit being returned from the synthesis could be anything. It should be checked whether the resulting circuit indeed implements the originally intended functionality, i.e., if you simulate the resulting circuit, the results should match the initial PLA.

Just to not forget: you might also want to set the constant inputs and garbage outputs on the resulting circuit object. The corresponding routines in the QFR are something like `qc.setLogicalQubitAncillary(q)` and `qc.setLogicalQubitGarbage(q)`.

Reply:
I still haven't implemented the decoder to check that. There should be some other way to right?

Reply:
You still have the encoded TT to check against. The circuit should realize the encoded TT.

Reply:
Yes, you are right. However, the encoded TT is not complete since the output consist of don't care care conditions (For e.g., please se the attached).
![Screenshot from 2022-11-26 12-10-36](https://user-images.githubusercontent.com/98882002/204085763-783e3f51-004d-4e4c-8e60-6a70dd268e30.png). How do I check the circuit synthesized with his type of pla?

Reply:
In the simplest fashion possible: take every possible input, simulate the circuit with that input and assert that the output matches the expected output (i.e., the values which are no don't cares match up).

Reply:
Okay. I was looking into the simulation feature offered by qfr. Is there a example of the simulation is checked?

Reply:
Since you are simulating a reversible circuit, a single simulation run or shot should be enough to determine the result.
For that you can use the simulation routine at
https://github.com/cda-tum/qfr/blob/e5f4189e09746d1a324901834a0b3bc0d87b2853/include/dd/Simulation.hpp#L39
with `shots=1`.
As for the initial state generation, you probably want to call 
https://github.com/cda-tum/dd_package/blob/8c1ca0622c29a3d2999d81caef1e0770cf7d7999/include/dd/Package.hpp#L266